coral-schema/src/main/java/com/linkedin/coral/schema/avro/RelToAvroSchemaConverter.java

```diff
@@ -200,8 +200,16 @@ public RelNode visit(LogicalProject logicalProject) {
       Schema inputSchema = schemaMap.get(logicalProject.getInput());
 
       Queue<String> suggestedFieldNames = new LinkedList<>();
-      for (RelDataTypeField field : logicalProject.getRowType().getFieldList()) {
-        suggestedFieldNames.offer(field.getName());
+
+      for (int i = 0; i < logicalProject.getChildExps().size(); ++i) {
+        final String fieldName = logicalProject.getRowType().getFieldList().get(i).getName();
+        if (fieldName.startsWith("EXPR$")) {
```

Comment:
I am not sure if this is the best way to go about the fix.
1. It is possible that some column is really named `EXPR$...` which will be renamed to `_c0`.
2. Some systems may use a different convention to generate names for unnamed columns. E.g. If you create the same view through Spark, the schema persisted in the metastore is a different. It as an escaped version of the expression itself.
```
hive (u_smahadik)> DESC FORMATTED v6;
OK
# col_name            	data_type           	comment

true                	boolean
lower(str)          	string
```
I was wondering if we should handle this at an earlier layer when reading view metadata from Hive.

Reply:
Thanks for your review and thoughts!
For 1, I tried on our internal Hive, we couldn't use `$` in column name while creating a table/view.
For 2, yeah it is an issue of this implementation. But to handle it at an earlier layer when reading view metadata from Hive, we might need big refactoring in coral-schema, because we only handle the base table (not view) metadata in `RelToAvroSchemaConverter` at present. Not sure if it's doable and necessary in terms of ROI.
@funcheetah @wmoustafa feel free to share your thoughts.

Reply:
You can use `$` in column names in Hive by using the backticks notation, e.g.
```
CREATE VIEW test AS
SELECT 1 AS `EXPR$1`
FROM t
```
At this point, I don't think this fix is super urgent for us. So I would be fine if the fix took some more time but considered more cases. But I will leave it to you guys to decide the best path forward.